nifi-extension-bundles/nifi-extension-utils/nifi-git-flow-registry/src/main/java/org/apache/nifi/registry/flow/git/client/GitRepositoryClient.java

```diff
@@ -0,0 +1,147 @@
+     * @throws IOException if an I/O error occurs
+     * @throws FlowRegistryException if a non-I/O error occurs
+     */
+    Set<String> getFileNames(String directory, String branch) throws IOException, FlowRegistryException;
+
+    /**
+     * Retrieves the commits for the file at the given path on the given branch.
+     *
+     * @param path the path of the file
+     * @param branch the branch name
+     * @return the list of commits
+     * @throws IOException if an I/O error occurs
+     * @throws FlowRegistryException if a non-I/O error occurs
+     */
+    List<GitCommit> getCommits(String path, String branch) throws IOException, FlowRegistryException;
+
+    /**
+     * Retrieves the content for the file at the given path on the given branch.
+     *
+     * @param path the path of the file
+     * @param branch the branch name
+     * @return the input stream for the content
+     * @throws IOException if an I/O error occurs
+     * @throws FlowRegistryException if a non-I/O error occurs
+     */
+    InputStream getContentFromBranch(String path, String branch) throws IOException, FlowRegistryException;
+
+    /**
+     * Retrieves the content for the file at the given path from the given commit.
+     *
+     * @param path the path of the file
+     * @param commitSha the commit for the content
+     * @return the input stream for the content
+     * @throws IOException if an I/O error occurs
+     * @throws FlowRegistryException if a non-I/O error occurs
+     */
+    InputStream getContentFromCommit(String path, String commitSha) throws IOException, FlowRegistryException;
+
+    /**
+     * Retrieves the SHA of the file at the given path on the given branch.
+     *
+     * @param path the path of the file
+     * @param branch the branch
+     * @return the SHA of the content
+     * @throws IOException if an I/O error occurs
+     * @throws FlowRegistryException if a non-I/O error occurs
+     */
+    Optional<String> getContentSha(String path, String branch) throws IOException, FlowRegistryException;
+
+    /**
+     * Creates a file in the repository based on the given request.
+     *
+     * @param request the request
+     * @return the SHA of the commit that created the file
+     * @throws IOException if an I/O error occurs
+     * @throws FlowRegistryException if a non-I/O error occurs
+     */
+    String createContent(GitCreateContentRequest request) throws IOException, FlowRegistryException;
+
+    /**
+     * Deletes the file at the given path on the given branch.
+     *
+     * @param filePath the path of the file
+     * @param commitMessage the commit message
+     * @param branch the branch
+     * @return the input stream to the deleted content
+     * @throws IOException if an I/O error occurs
+     * @throws FlowRegistryException if a non-I/O error occurs
+     */
+    InputStream deleteContent(String filePath, String commitMessage, String branch) throws FlowRegistryException, IOException;
```

Comment:
It seems like the caller is responsible to close the stream? That would be worth noting one way or the other.